internal/metric/global/delegate_instruments.go

```diff
@@ -0,0 +1,458 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package global // import "go.opentelemetry.io/otel/internal/metric/global"
+
+import (
+	"context"
+	"sync/atomic"
+
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/metric/sdkapi"
+)
+
+var noopFloat64Counter, _ = noopMeter.NewFloat64Counter("")
+var noopInt64Counter, _ = noopMeter.NewInt64Counter("")
+var noopFloat64UpDownCounter, _ = noopMeter.NewFloat64UpDownCounter("")
+var noopInt64UpDownCounter, _ = noopMeter.NewInt64UpDownCounter("")
+var noopFloat64Histogram, _ = noopMeter.NewFloat64Histogram("")
+var noopInt64Histogram, _ = noopMeter.NewInt64Histogram("")
+
+var noopFloat64GaugeObserver, _ = noopMeter.NewFloat64GaugeObserver("", nil)
+var noopInt64GaugeObserver, _ = noopMeter.NewInt64GaugeObserver("", nil)
+var noopFloat64CounterObserver, _ = noopMeter.NewFloat64CounterObserver("", nil)
+var noopInt64CounterObserver, _ = noopMeter.NewInt64CounterObserver("", nil)
+var noopFloat64UpDownCounterObserver, _ = noopMeter.NewFloat64UpDownCounterObserver("", nil)
+var noopInt64UpDownCounterObserver, _ = noopMeter.NewInt64UpDownCounterObserver("", nil)
+
+type float64CounterImpl struct {
+	name     string
+	options  []metric.InstrumentOption
+	delegate atomic.Value // metric.Float64Counter
```

Comment:
Is there any way for there to be different concrete types stored in this value?  If so this approach won't work: https://pkg.go.dev/sync/atomic#Value.Store .  If not then I think we should change these comments to the concrete type we are storing, so as to not leave the impression that we can safely use anything that implements this interface.

Reply:
Agreed, I will look into this when the other 2 PRs that this depends on are merged. I crafted this mostly to show that we can remove the need of everything using sdkapi.